app/models/mixins/saved_url_mixin.rb

```diff
@@ -0,0 +1,15 @@
+module SavedUrlMixin
+  extend ActiveSupport::Concern
+
+  def saved_url=(url, userid)
+    # , :userid => userid).each(&:destroy)
+    SavedUrl.where(:refers_to_id => id, :refers_to_type => self.class.name).each(&:destroy)
```

Comment:
This could be handled as a polymorphic relationship and would be something like:

```ruby
SavedUrl.where(:resource => self).destroy_all
```

But more likely would just be one of these depending on if it is a `has_one` or `has_many`:
`saved_url.destroy`
or
`saved_urls.destroy_all`